### PR TITLE
Jacob/feat/postform

### LINKED ISSUE #18: Create new post form component
Literally just the form for adding new posts

### PR DESCRIPTION
**Pull request by Goldenaries658**
## Basic form using mui components and CKEditor
- **Inputs**
  * These are controlled inputs which edit state onChange, will then be sending the state off with click of fab.
- **FABs**
  * Added two FABs, on on `'/'` route to link to postform, and another that will be used to submit data.
  * These animate and change colours.
- **Checkboxes**
  * Populated using state, will need to manipulate state before sending to server to only include options that are set to `true`
  * Only allows the user to select a max of two items, is currently set as required so shouldn't allow them to skip it either.
- **CKEditor**
  * Also saves to state, will save in formatted html so super easy to use later on pages as we can just spit out the html straight up.
  * This starts with a default height but will expand up to a max point.
- **AutoComplete (chips)**
  * Takes in an array of objects with options, I have added this as a json file alongside the component, but will need to go into db at some point so it is always uniform across all versions of the app.
  * Doesn't allow a user to enter anything other than the predefined choices.
  * The currently selected option will save directly to state.

### CODE REVIEW

File: src/pages/NewPost/ErrorAlert.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      <Dialog
+        open={isAlertOpen}
+        onClose={handleClose}
+        aria-labelledby="Error"
+        aria-describedby="There is something wrong."
+      >
+        <DialogContent>
+          <DialogContentText id="alert-dialog-description">
+            {`You need to add ${
+              errorKey === 'categories' || errorKey === 'something' ? '' : 'a '
```
Comment (phillidp1989) [inline src/pages/NewPost/ErrorAlert.js]: Really like the error handling. The only thing I would suggest is separate messages to handle zero categories checked and too many categories checked.

**Review by phillidp1989 — COMMENTED** [review]
Excellent. I think it looks great for a 'simple' form. Nice and responsive, UX is great. Form is quick and simple to complete.

Just a couple of suggestions - could you add in the user context and only display the 'add' FAB when the user is logged in? Potentially show something different to indicate a user must be logged in to add a post. Also, when the user state is available, we will need to send the user's ID along with the post data when populating the DB.

**Comment by Goldenaries658** [comment]
RE: The add button on the main page, I purposely left that in with redirection to the login page, to entice users to sign up. Marketing strategy

File: src/pages/NewPost/ErrorAlert.js (hunk 1 above)
Comment (Goldenaries658) [reply src/pages/NewPost/ErrorAlert.js]: Done, error messages and handling largely improved to deal with much more scenarios

**Review by phillidp1989 — APPROVED** [review]

**Pull request merged**